tests/plugins/test_accounts.py

```diff
@@ -0,0 +1,219 @@
+        self.app.post_json('/accounts', {'data': {'id': 'me', 'password': 'bouh'}},
+                           status=201)
+        resp = self.app.get('/', headers=get_user_headers('me', 'bouh'))
+        assert resp.json['user']['id'] == 'account:me'
+
+    def test_password_field_is_mandatory(self):
+        self.app.post_json('/accounts', {'data': {'id': 'me'}}, status=400)
+
+    def test_account_can_have_metadata(self):
+        resp = self.app.post_json('/accounts',
+                                  {'data': {'id': 'me', 'password': 'bouh', 'age': 42}},
+                                  status=201)
+        assert resp.json['data']['age'] == 42
+
+    def test_cannot_create_account_if_already_exists(self):
+        self.app.post_json('/accounts', {'data': {'id': 'me', 'password': 'bouh'}},
+                           status=201)
+        resp = self.app.post_json('/accounts', {'data': {'id': 'me', 'password': 'bouh'}},
+                                  status=403)
+        assert 'already exists' in resp.json['message']
+
+    def test_username_and_account_id_must_match(self):
+        resp = self.app.put_json('/accounts/alice', {'data': {'id': 'bob', 'password': 'bouh'}},
+                                 status=400)
+        assert 'does not match' in resp.json['message']
+
+    def test_returns_existing_account_if_authenticated(self):
+        self.app.post_json('/accounts', {'data': {'id': 'me', 'password': 'bouh'}},
+                           status=201)
+        self.app.post_json('/accounts', {'data': {'id': 'me', 'password': 'bouh'}},
+                           headers=get_user_headers('me', 'bouh'),
+                           status=200)
+
+    def test_cannot_create_other_account_if_authenticated(self):
+        self.app.post_json('/accounts', {'data': {'id': 'me', 'password': 'bouh'}},
+                           status=201)
+        resp = self.app.post_json('/accounts', {'data': {'id': 'you', 'password': 'bouh'}},
+                                  headers=get_user_headers('me', 'bouh'),
+                                  status=400)
+        assert 'do not match' in resp.json['message']
+
+
+class AccountUpdateTest(AccountsWebTest):
+
+    def setUp(self):
+        self.app.put_json('/accounts/alice', {'data': {'password': '123456'}}, status=201)
+
+    def test_password_can_be_changed(self):
+        self.app.put_json('/accounts/alice', {'data': {'password': 'bouh'}},
+                          headers=get_user_headers('alice', '123456'),
+                          status=200)
+
+    def test_authentication_with_old_password_is_denied_after_change(self):
+        self.app.put_json('/accounts/alice', {'data': {'password': 'bouh'}},
+                          headers=get_user_headers('alice', '123456'),
+                          status=200)
+        self.app.get('/accounts/alice', headers=get_user_headers('alice', '123456'),
+                     status=401)
```

Comment:
I'm not sure about 401 here, this can leak information about the account existing. Maybe making it 403?